app/views/appointment_summaries/_preview_form.html.erb

```diff
@@ -7,6 +7,7 @@
   <% end %>
 
   <%= f.submit local_assigns.fetch(:submit_button_text),
-    class: (%w(btn btn-lg) | Array(local_assigns[:submit_button_classes])) %>
+    class: (%w(btn btn-lg) | Array(local_assigns[:submit_button_classes])),
+    onclick: 'this.disabled = true; this.parentElement.submit();' %>
```

Comment:
Yeah, I was thinking of where to put the code and this was the simplest possible thing that would work. Hence my apprehension.